src/engraving/dom/segment.cpp

```diff
@@ -2395,6 +2395,16 @@ void Segment::createShapes()
     addPreAppendedToShape();
 }
 
+static PointF staffOffset(EngravingItem* e)
```

Comment:
This definitely shouldn't be in the Segment class, as it's got nothing to do with segments. It should really be a method of EngravingItem. Something like:
```
double EngravingItem::staffOffset() const 
{
    const StaffType* st = staffType();
    double yOffset = st ? st->yOffset().val() * spatium() : 0.0;
    return PointF(0.0, yOffset);
}
```
and perhaps you could also add in engravingitem.h something like
`double yStaffOffset() const { return staffOffset().y(); }`
for convenience when you just need the y.

This will allow you to call `item->staffOffset()` or `item->yStaffOffset()` on any EngravingItem, anywhere you need it, and will save you from repeating those two lines of code a million times ;)